src/main/kotlin/org/jitsi/nlj/rtp/TransportCcEngine.kt

```diff
@@ -142,22 +156,69 @@ class TransportCcEngine(
         }
     }
 
+    /** Trim old sent packet details from the map.  Caller should be synchronized on [sentPacketsSyncRoot]. */
+    private fun cleanupSentPacketDetails(newSeq: Int): Boolean {
+        if (sentPacketDetails.isEmpty()) {
+            return true
+        }
+
+        val latestSeq = sentPacketDetails.lastKey()
+
+        /* If our sequence numbers have jumped by a quarter of the number space, reset the map. */
+        val seqJump = getSequenceNumberDelta(newSeq, latestSeq)
+        if (seqJump >= 0x4000 || seqJump <= -0x4000) {
+            sentPacketDetails.clear()
+            return true
+        }
+
+        val threshold = applySequenceNumberDelta(latestSeq, -MAX_OUTGOING_PACKETS_HISTORY)
+
+        if (isOlderSequenceNumberThan(newSeq, threshold)) {
+            return false
+        }
+
+        val it = sentPacketDetails.navigableKeySet().iterator()
+        while (it.hasNext()) {
+            val key = it.next()
+            if (isOlderSequenceNumberThan(key, threshold)) {
+                it.remove()
+            } else {
+                break
+            }
+        }
+
+        return true
+    }
+
     fun mediaPacketSent(tccSeqNum: Int, length: DataSize) {
         synchronized(sentPacketsSyncRoot) {
             val now = clock.instant()
-            sentPacketDetails.put(
-                tccSeqNum and 0xFFFF,
-                PacketDetail(length, now))
+            val seq = tccSeqNum and 0xFFFF
+            if (cleanupSentPacketDetails(seq)) {
+                sentPacketDetails.put(seq, PacketDetail(length, now))
+            }
         }
     }
 
+    /**
+     * [PacketDetailState] is the state of a [PacketDetail]
+     */
+    private enum class PacketDetailState {
+        unreported, reportedLost, reportedReceived
+    }
+
     /**
      * [PacketDetail] is an object that holds the
      * length(size) of the packet in [packetLength]
      * and the time stamps of the outgoing packet
      * in [packetSendTime]
      */
-    private data class PacketDetail internal constructor(internal val packetLength: DataSize, internal val packetSendTime: Instant)
+    private data class PacketDetail internal constructor(
+        internal val packetLength: DataSize,
+        internal val packetSendTime: Instant
+    ) {
+        var state = PacketDetailState.unreported
```

Comment:
Can we put a comment here?  If I'm understanding it right, it's something like:

```
[state] represents the state of this packet detail with regards to the reception of a TCC feedback from the remote side.  [PacketDetail]s start out as [unreported]: once we receive a TCC feedback from the remote side referring to this packet, the state will transition to either [reportedLost] or [reportedReceived].
```
?

Reply:
Ok.